src/test/java/org/kie/processmigration/test/MockKieServerLifecycleManager.java

```diff
@@ -0,0 +1,262 @@
+        kieservers.put("kieservers[0].username", "admin");
+        kieservers.put("kieservers[0].password", "admin123");
+        kieservers.put("kieservers[1].host", server2Response.getResult().getLocation());
+        kieservers.put("kieservers[1].username", "admin");
+        kieservers.put("kieservers[1].password", "admin123");
+        kieservers.put("kieservers[2].host", wireMockServer.baseUrl() + "/not-found/services/rest/server");
+        kieservers.put("kieservers[2].username", "unused");
+        kieservers.put("kieservers[2].password", "unused");
+        kieservers.put("kieservers[3].host", wireMockServer.baseUrl() + "/unauthorized/services/rest/server");
+        kieservers.put("kieservers[3].username", "unused");
+        kieservers.put("kieservers[3].password", "unused");
+        kieservers.put("kieservers[4].host", wireMockServer.baseUrl() + "/forbidden/services/rest/server");
+        kieservers.put("kieservers[4].username", "other");
+        kieservers.put("kieservers[4].password", "other123");
+        kieservers.put("kieservers[5].host", wireMockServer.baseUrl() + "/retry/services/rest/server");
+        kieservers.put("kieservers[5].username", "other");
+        kieservers.put("kieservers[5].password", "other123");
+        return kieservers;
+    }
+
+    private void stubContainers() throws JsonProcessingException {
+        ServiceResponse<KieContainerResourceList> emptyContainersResponse = new ServiceResponse<>(
+                KieServiceResponse.ResponseType.SUCCESS,
+                "List of created containers",
+                new KieContainerResourceList());
+        stubFor(get(urlPathEqualTo("/kie-server-1/services/rest/server/containers"))
+                .withBasicAuth("admin", "admin123")
+                .willReturn(okJson(mapper.writeValueAsString(emptyContainersResponse))));
+
+        KieContainerResourceList containers = new KieContainerResourceList();
+        containers.getContainers().add(new KieContainerResource("example-1.0.0", new ReleaseId("com.myspace", "Example", "1.0.0"), KieContainerStatus.STARTED));
+        containers.getContainers().add(new KieContainerResource("example-1.0.1", new ReleaseId("com.myspace", "Example", "1.0.1"), KieContainerStatus.STARTED));
+        ServiceResponse<KieContainerResourceList> containersResponse = new ServiceResponse<>(
+                KieServiceResponse.ResponseType.SUCCESS,
+                "List of created containers",
+                containers);
+        stubFor(get(urlPathEqualTo("/kie-server-2/services/rest/server/containers"))
+                .withBasicAuth("admin", "admin123")
+                .willReturn(okJson(mapper.writeValueAsString(containersResponse))));
+
+        ProcessDefinition process1 = new ProcessDefinition();
+        process1.setContainerId("example-1.0.0");
+        process1.setId("process1");
+        process1.setVersion("1.0");
+        process1.setPackageName("org.kie.examples");
+        process1.setName("The process 1");
```

Comment:
Maybe you can create a method to avoid duplicating the code for creating the process definitions objects